code/game/objects/effects/mines.dm

```diff
@@ -11,42 +11,56 @@
 	var/triggered = 0
 
 /obj/effect/mine/New()
+	..()
 	icon_state = "uglyminearmed"
 
-/obj/effect/mine/Crossed(AM as mob|obj)
-	Bumped(AM)
+/obj/effect/mine/Crossed(mob/living/carbon/AM)
+	if(istype(AM))
+		visible_message("<font color='red'>[AM] triggered the [bicon(src)] [src]</font>")
+		trigger(AM)
 
-/obj/effect/mine/Bumped(mob/M as mob|obj)
-
-	if(triggered) return
+/obj/effect/mine/proc/trigger(mob/living/carbon/AM)
+	explosion(loc, 0, 1, 2, 3)
+	qdel(src)
 
-	if(istype(M, /mob/living/carbon/human) || istype(M, /mob/living/carbon/monkey))
-		for(var/mob/O in viewers(world.view, src.loc))
-			to_chat(O, "<font color='red'>[M] triggered the [bicon(src)] [src]</font>")
-		triggered = 1
-		call(src,triggerproc)(M)
+/obj/effect/mine/dnascramble
+	name = "Radiation Mine"
 
-/obj/effect/mine/proc/triggerrad(obj)
+/obj/effect/mine/dnascramble/proc/trigger(mob/living/carbon/AM)
 	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
 	s.set_up(3, 1, src)
 	s.start()
-	obj:radiation += 50
-	randmutb(obj)
-	domutcheck(obj,null)
-	spawn(0)
-		qdel(src)
-
-/obj/effect/mine/proc/triggerstun(obj)
-	if(ismob(obj))
-		var/mob/M = obj
-		M.Stun(30)
+	AM.radiation += 50
+	randmutb(AM)
+	domutcheck(AM,null)
+	qdel(src)
+
+/obj/effect/mine/plasma
+	name = "Plasma Mine"
+
+/obj/effect/mine/plasma/proc/trigger(AM)
```

Comment:
That was what caused the compiler error, it was a leftover from when I converted the individual procs under the same name trigger.